content/docs/1.11.0/advanced-resources/deploy/csi-pod-antiaffinity-preset.md

```diff
@@ -0,0 +1,52 @@
+---
+title: CSI Component Pod Anti-Affinity
+weight: 5
+---
+
+This document describes how to configure pod anti-affinity for Longhorn's CSI components. This feature enhances the resilience of the storage system, particularly in smaller clusters, by preventing multiple replicas of a CSI component from running on the same node and thereby avoiding a single point of failure.
+
+For more information on pod anti-affinity, refer to the official Kubernetes documentation on [Inter-pod affinity and anti-affinity](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#inter-pod-affinity-and-anti-affinity).
+
+## Configuring Pod Anti-Affinity for Longhorn CSI Components
+
+You can configure the pod anti-affinity for the following Longhorn CSI components:
+
+- `csi-attacher`
+- `csi-provisioner`
+- `csi-resizer`
+- `csi-snapshotter`
+
+The `podAntiAffinityPreset` setting has two available values:
+
+- **`soft` (default)**: This is a "best-effort" rule using `preferredDuringSchedulingIgnoredDuringExecution`. The scheduler will try to prevent multiple CSI component replicas from running on the same node, but it is not guaranteed.
+- **`hard`**: This is a "strict" rule using `requiredDuringSchedulingIgnoredDuringExecution`. The scheduler will not schedule a pod on a node if it violates the anti-affinity rule. This may result in pods remaining in a `Pending` state if no suitable nodes are available.
+
+### During Longhorn Installation
```

Comment:
How about day 2? If users want to update the settings after installation or even upgrade from a previous version. What's the behavior? Suggest briefly explaining that.

Reply:
Currently, the only way to update these settings is by manually deleting metadata (such as `driver.longhorn.io/git-commit` or `driver.longhorn.io/version`) from the CSI deployments to force a redeploy, and then editing the flag in `longhorn-driver-deployer`.
A better approach would be to automatically monitor the flag change to execute these alterations, which is why I opened a new Issue and PR for this: https://github.com/longhorn/longhorn/issues/12100